projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -1,140 +1,330 @@
+            it('should set label prop in dot-label', async () => {
+                element.setProperty('label', 'Name:');
+                await page.waitForChanges();
+                const label = await dotTestUtil.getDotLabel(page);
+                expect(await label.getAttribute('label')).toBe('Name:');
+            });
+            it('should set label prop in dot-label when is a unexpected value', async () => {
+                element.setProperty('label', [1, 2, 'test']);
+                await page.waitForChanges();
+                const label = await dotTestUtil.getDotLabel(page);
+                expect(await label.getAttribute('label')).toEqual('1,2,test');
+            });
+        });
+
+        describe('placeholder', () => {
+            it('should set placeholder correctly', async () => {
+                element.setProperty('placeholder', 'Test');
+                await page.waitForChanges();
+                expect(input.getAttribute('placeholder')).toBe('Test');
+            });
+            it('should set placeholder correctly with invalid value', async () => {
+                element.setProperty('placeholder', { test: 'hi' });
+                await page.waitForChanges();
+                expect(input.getAttribute('placeholder')).toBe('[object Object]');
+            });
+        });
+
+        describe('hint', () => {
+            it('should set hint correctly', async () => {
+                element.setProperty('hint', 'Test');
+                await page.waitForChanges();
+                expect((await dotTestUtil.getHint(page)).innerText).toBe('Test');
+            });
+
+            it('should not render hint', async () => {
+                expect(await dotTestUtil.getHint(page)).toBeNull();
+            });
+
+            it('should not render  hint with invalid value', async () => {
+                element.setProperty('hint', { test: 'hint' });
+                await page.waitForChanges();
+                expect(await dotTestUtil.getHint(page)).toBeNull();
+            });
+        });
+
+        describe('required', () => {
+            it('should render required attribute and dot-required class', async () => {
+                element.setProperty('required', 'true');
+                await page.waitForChanges();
+                expect(input.getAttribute('required')).toBeDefined();
+                expect(element).toHaveClasses(['dot-required']);
+            });
+
+            it('should not render required attribute', async () => {
+                element.setProperty('required', 'false');
+                await page.waitForChanges();
+                expect(input.getAttribute('required')).toBeNull();
+            });
+
+            it('should render required attribute with invalid value', async () => {
+                element.setProperty('required', { test: 'test' });
+                await page.waitForChanges();
+                expect(input.getAttribute('required')).toBeDefined();
```

Comment:
Where do you tests the default required message?

Reply:
updated, here, https://github.com/dotCMS/core-web/pull/1032/files#diff-5fdc3173364322f489e7fb5fd2408fafR178